include/ccf/crypto/cose_signer.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/crypto/pem.h"
+
+namespace ccf::crypto
+{
+  class COSESigner
+  {
+  public:
+    virtual bool sign(
+      std::span<uint8_t> payload,
+      std::vector<uint8_t>& cose_envelope) const = 0;
```

Comment:
I can't see an obvious reason this is passed by mutable ref, and the bool return is not a very expressive or extensible way to expose errors. I suggest we return the cose_envelope, and throw if there is an error.